### PR TITLE
Develop avm freeze

### PR DESCRIPTION
**Pull request by ealdrich-figure**
First lien flag is added back to the cltv decline function. The booleans for deciles were also modified to intersect with only the approved population.

### CODE REVIEW

File: avm_freeze/main.py [hunk 1]
```diff
@@ -20,17 +20,20 @@
...
-bool_bottom_decile_cltv = data.pre_loan_adj_cltv < data.pre_loan_adj_cltv.loc[bool_approved].quantile(0.9)
-bool_bottom_decile_dti = data.pre_loan_dti < data.pre_loan_dti.loc[bool_approved].quantile(0.9)
+bool_bottom_decile_fico = bool_approved\
+                          & (data.credit_soft_fico < data.credit_soft_fico.loc[bool_approved].quantile(0.1))
+bool_top_decile_cltv = bool_approved\
+                       & (data.pre_loan_adj_cltv < data.pre_loan_adj_cltv.loc[bool_approved].quantile(0.9))
+bool_top_decile_dti = bool_approved & (data.pre_loan_dti < data.pre_loan_dti.loc[bool_approved].quantile(0.9))
 bool_net = (bool_approved | bool_swap_in) & ~bool_swap_out
 
 # %% Compute credit attributes for a population (sorry for this ugliness, Jakub...didn't have a chance to clean)
```
Comment (jjurek-figure) [inline avm_freeze/main.py]: nit :: remove the parenthetical remark in comment

File: avm_freeze/avm_freeze.py [hunk 2]
```diff
@@ -10,11 +10,12 @@ def compute_adjusted_avm(avm_values, fsd_values, avm_cutoff, fsd_cutoff):
...
-    bool_group_1 = (fico_values < 680) & (cltv_adj_values > 0.75)
-    bool_group_2 = (fico_values >= 680) & (fico_values < 720) & (cltv_adj_values > 0.9)
-    bool_group_3 = (fico_values >= 720) & (cltv_adj_values > 0.95)
-    cltv_decline_bool = bool_group_1 | bool_group_2 | bool_group_3
+def compute_cltv_decline(fico_values, cltv_adj_values, first_lien_values):
+    bool_group_1 = (fico_values < 680) & (cltv_adj_values > 0.75) & (~first_lien_values)
+    bool_group_2 = (fico_values >= 680) & (fico_values < 720) & (cltv_adj_values > 0.9) & (~first_lien_values)
+    bool_group_3 = (fico_values >= 720) & (cltv_adj_values > 0.95) & (~first_lien_values)
+    bool_group_4 = (cltv_adj_values > 0.50) & first_lien_values
+    cltv_decline_bool = bool_group_1 | bool_group_2 | bool_group_3 | bool_group_4
```
Comment (jjurek-figure) [inline avm_freeze/avm_freeze.py]: looks good. +1

File: avm_freeze/main.py [hunk 3]
```diff
@@ -20,17 +20,20 @@
...
 # %% Approved/Swap-in/Swap-out populations
 bool_avm_freeze_window = (pd.to_datetime(data.app_start_date)
                           - pd.to_datetime(data['Geo AVM Value Date'])).dt.days < AVM_FREEZE_RANGE
 bool_approved = data.product_selected_flag == 'Y'
 bool_swap_out = bool_avm_freeze_window & bool_approved & data.cltv_frozen_decline_flag
 bool_swap_in = bool_avm_freeze_window & (data.decline_type == 'CLTV') & (~data.cltv_frozen_decline_flag)
-bool_bottom_decile_fico = data.credit_soft_fico < data.credit_soft_fico.loc[bool_approved].quantile(0.1)
-bool_bottom_decile_cltv = data.pre_loan_adj_cltv < data.pre_loan_adj_cltv.loc[bool_approved].quantile(0.9)
-bool_bottom_decile_dti = data.pre_loan_dti < data.pre_loan_dti.loc[bool_approved].quantile(0.9)
+bool_bottom_decile_fico = bool_approved\
```
Comment (jjurek-figure) [inline avm_freeze/main.py]: Yes. Good catch. We only care about these characteristics among the baseline population of *approved* applicants. This should remove the observations with the unusual FICOs

File: avm_freeze/main.py [hunk 4]
```diff
@@ -20,17 +20,20 @@
                                                     avm_cutoff=AVM_CUTOFF,
                                                     fsd_cutoff=FSD_CUTOFF)
 data['cltv_adj_frozen'] = data.pre_loan_adj_cltv*data.avm_adj_application_soft/data.avm_adj_selection
```
Comment (jjurek-figure) [inline avm_freeze/main.py]: nit: spaces around binary operators

Comment (ealdrich-figure) [reply avm_freeze/main.py]: I think this conforms with PEP 8:

Always surround these binary operators with a single space on either side: assignment (=), augmented assignment (+=, -= etc.), comparisons (==, <, >, !=, <>, <=, >=, in, not in, is, is not), Booleans (and, or, not).

**Pull request merged**